packages/glimmer-syntax/lib/parser/handlebars-node-visitors.ts

```diff
@@ -128,6 +128,18 @@ export default {
   },
 
   PathExpression: function(path) {
+    let { original } = path;
+
+    if (original[0] === '@') {
+      original = original.slice(1);
+    }
+
+    if (path.depth > 0) {
+      throw new Error(`Context switch detected. Glimmer does not support context switching, you should remove the context switching on ${path.original}`);
```

Comment:
I'm not sure about this error message, also, we should probably assert `{{./foo}}` which is essentially `{{this.foo}}` I'm not sure what we are doing with that.